metrics/resetting_timer.go

```diff
@@ -53,27 +53,20 @@ func (t *ResettingTimer) Snapshot() *ResettingTimerSnapshot {
 	return snapshot
 }
 
-// Time records the duration of the execution of the given function.
-func (t *ResettingTimer) Time(f func()) {
-	ts := time.Now()
-	f()
-	t.Update(time.Since(ts))
-}
-
 // Update records the duration of an event.
-func (t *ResettingTimer) Update(d time.Duration) {
+func (t *ResettingTimer) Update(d int64) {
```

Comment:
Why change the type here to `int64`? It seems more correct to use `time.Duration`, and the underlying type is the same.